src/main/java/me/moodcat/backend/rooms/RoomInstance.java

```diff
@@ -206,7 +217,8 @@ private void processVotes(final Song previousSong) {
     private void processNextSong(final Room room, final List<Song> history) {
         final List<Song> playQueue = room.getPlayQueue();
         if (playQueue.isEmpty()) {
-            playQueue.addAll(history);
+            VAVector vector = this.roomDAOProvider.get().findById(id).getVaVector();
+            playQueue.addAll(this.songDAOProvider.get().findForDistance(vector, NUMBER_OF_SELECTED_SONGS));
```

Comment:
If we want to keep the repeat flag for a while, either as a development tool or a feature - for example as a room command or something, we might want to use this behaviour only when repeating is disabled, and fall back to the history when repeat is enabled.